go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -333,10 +333,16 @@ func (oc *Controller) deleteEgressFirewall(egressFirewallObj *egressfirewallapi.
 	deleteDNS := false
 	obj, loaded := oc.egressFirewalls.LoadAndDelete(egressFirewallObj.Namespace)
 	if !loaded {
-		return fmt.Errorf("there is no egressFirewall found in namespace %s", egressFirewallObj.Namespace)
+		return fmt.Errorf("there is no egressFirewall found in namespace %s",
+			egressFirewallObj.Namespace)
 	}
 
-	ef, _ := obj.(egressFirewall)
+	ef, ok := obj.(*egressFirewall)
+	if !ok {
+		return fmt.Errorf("type assertion to *egressFirewall failed for "+
```

Comment:
I can't see from the code how this would ever happen, given that the only thing writing to `oc.egressfirewalls` is addEgressFirewall() calling cloneEgressFirewall() which always returns the expected object type. But if you want to keep this, please add a `%T` somwhere in there so we know what incorrect type `ef` acutally was.

Reply:
Sure! I just pushed an improved error message as suggested. Thanks for the feedback!